integrations/va-auth0/src/index.tsx

```diff
@@ -0,0 +1,251 @@
+            </block>
+        );
+    },
+});
+
+const handleFetchEvent: FetchEventCallback<Auth0RuntimeContext> = async (request, context) => {
+    const { environment } = context;
+    const installationURL = environment.spaceInstallation?.urls?.publicEndpoint;
+    if (installationURL) {
+        const router = Router({
+            base: new URL(installationURL).pathname,
+        });
+
+        router.get('/visitor-auth/response', async (request) => {
+            if (context.environment.spaceInstallation?.space) {
+                const space = await context.api.spaces.getSpaceById(
+                    context.environment.spaceInstallation?.space
+                );
+                const spaceData = space.data;
+                const privateKey = context.environment.signingSecret;
+                let token;
+                try {
+                    token = await sign(
+                        { exp: Math.floor(Date.now() / 1000) + 2 * (60 * 60) },
+                        privateKey
+                    );
+                } catch (e) {
+                    return new Response('Error: Could not sign JWT token', {
+                        status: 500,
+                    });
+                }
+
+                const issuerBaseUrl = environment.spaceInstallation?.configuration.issuer_base_url;
+                const clientId = environment.spaceInstallation?.configuration.client_id;
+                const clientSecret = environment.spaceInstallation?.configuration.client_secret;
+                if (clientId && clientSecret) {
+                    const searchParams = new URLSearchParams({
+                        grant_type: 'authorization_code',
+                        client_id: clientId,
+                        client_secret: clientSecret,
+                        code: `${request.query.code}`,
+                        redirect_uri: `${installationURL}/visitor-auth/response`,
+                    });
+                    const url = `${issuerBaseUrl}/oauth/token/`;
+                    const resp: any = await fetch(url, {
+                        method: 'POST',
+                        headers: { 'content-type': 'application/x-www-form-urlencoded' },
+                        body: searchParams,
+                    })
+                        .then((response) => response.json())
+                        .catch((err) => {
+                            return new Response('Error: Could not fetch access token from Auth0', {
+                                status: 401,
+                            });
+                        });
+
+                    if ('access_token' in resp) {
+                        let url;
+                        if (request.query.state) {
+                            url = `${spaceData.urls?.published}${request.query.state}/?jwt_token=${token}`;
+                        } else {
+                            url = `${spaceData.urls?.published}/?jwt_token=${token}`;
```

Comment:
`request.query.state` contains the path the user wanted to go to before the auth flow started. We pass this path to the auth provider when the auth flow starts and the auth provider returns us this path when the auth flow completes.

So if a user tries to access a link behind VA, say: `docs.gitbook.com/my/docs`, we want the user to end up on the same link after authentication instead of redirecting them to `docs.gitbook.com` after authentication. 

`request.query.state` would be `my/docs` in this case.